BitwardenShared/Core/Platform/Utilities/DefaultValue.swift

```diff
@@ -0,0 +1,99 @@
+import OSLog
+
+// MARK: - DefaultValueProvider
+
+/// A protocol for defining a default value for a `Decodable` type if an invalid or missing value
+/// is received.
+///
+protocol DefaultValueProvider: Decodable {
+    /// The default value to use if the value to decode is invalid or missing.
+    static var defaultValue: Self { get }
+}
+
+// MARK: - DefaultValue
+
+/// A property wrapper that will default the wrapped value to a default value if decoding fails.
+/// This is useful for decoding types which may not be present in the response or to prevent a
+/// decoding failure if an invalid value is received.
+///
+@propertyWrapper
+struct DefaultValue<T: DefaultValueProvider> {
+    // MARK: Properties
+
+    /// The wrapped value.
+    let wrappedValue: T
+}
+
+// MARK: - Decodable
+
+extension DefaultValue: Decodable {
+    init(from decoder: any Decoder) throws {
+        let container = try decoder.singleValueContainer()
+        do {
+            wrappedValue = try container.decode(T.self)
+        } catch {
+            if let intValue = try? container.decode(Int.self) {
+                Logger.application.warning(
+                    """
+                    Cannot initialize \(T.self) from invalid Int value \(intValue), \
+                    defaulting to \(String(describing: T.defaultValue)).
+                    """
+                )
+            } else if let stringValue = try? container.decode(String.self) {
+                Logger.application.warning(
+                    """
+                    Cannot initialize \(T.self) from invalid String value \(stringValue), \
```

Comment:
🤔 I think we should be using [OSLogPrivacy private](https://developer.apple.com/documentation/os/oslogprivacy/3578096-private) instead of the default [auto](https://developer.apple.com/documentation/os/oslogprivacy/3578094-auto) here where the system decides whether to show or redact the `stringValue`. Or at least choose which one to use depending on whether one is on debugging.
IMO it's more important here than the int one as we should always hide string values from logs when we don't actually know if it's safe to potentially be displayed.

Reply:
My understanding is that strings are always hidden when interpolated but I can specify the privacy level.

Reply:
That's my understanding as well, but I don't think there's any harm in explicitly specifying it, especially if someone is combing through the code later and doesn't know about the default being "hide".